meta-lxatac-bsp: rauc: split rauc.bb and rauc-conf.bb

hnez opened the pull request:
This PR is made in anticipation of a change in rauc/meta-rauc#147, which splits the rauc recipe into a recipe for the binaries and one for the configuration.

TODO before merging

- [x] Wait for the change to land in `meta-rauc` and change the submodule URL back to the upstream project.

hnez committed:
meta-lxatac-bsp: rauc: follow meta-rauc and split rauc.bb and rauc-conf.bb

This reduces how often rauc has to be recompiled.
And it also makes the main rauc recipe machine-independent.
This does not help us too much because we only have one machine in this
BSP, but it should be nice for others.

Split our .bbappends into one for rauc itself and one for the config.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

ejoerns commented on meta-lxatac-bsp/recipes-core/rauc/rauc-conf.bbappend:
```diff
@@ -0,0 +1,52 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
+
+DEPENDS:append = "openssl-native"
+RDEPENDS:${PN}:append = "bash"
```
As long as no overrides are used, I'd use simple `+=` for `DEPENDS` and `RDEPENDS` since they are easier to parse and this is consistent with other recipes. However, for the scope of this PR I'd leave it as it was before the split-up, anyway. (Same for `SRC_URI` below.)

hnez committed:
meta-lxatac-bsp: rauc: use simple ... += instead of ...:append =

The += syntax is easier to parse for bitbake, but also for human readers,
because the rules around it are a bit simpler than those for :append.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-rauc: update to revision with rauc release 1.11.2

This version of RAUC solves an issue we had to patch around where
the info command would not work (on the command line and via DBus).
This means we can drop the patch.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

ejoerns commented on meta-lxatac-bsp/recipes-core/rauc/rauc_%.bbappend:
```diff
@@ -1,8 +1,8 @@
 FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
 
-RDEPENDS:${PN}:append = "e2fsprogs-resize2fs"
+RDEPENDS:${PN} += " e2fsprogs-resize2fs"
```
Spaces at the beginning of a string are not required when using "+=".
In fact, they were required before when using `:append` :wink:

hnez replied:
\*sigh\*. Fixed

hnez commented:
I've just checked if this PR builds and that `rauc info` and update notifications work with rauc 1.11.2.
Look good, thanks for the release and review!
